storage/remote/queue_manager_test.go

```diff
@@ -321,20 +322,16 @@ func TestCalculateDesiredsShards(t *testing.T) {
 		m.numShards = c.startingShards
 		m.samplesIn.incr(c.samplesIn)
 		m.samplesOut.incr(c.samplesOut)
-		m.lastSendTimestamp = time.Now().Unix()
+		m.lastSendTimestamp = c.lastSendTimestamp
 
-		// Resharding shouldn't take place if the last successful send was > batch send deadline*2 seconds ago.
-		if !c.reshard {
-			m.lastSendTimestamp = m.lastSendTimestamp - int64(3*time.Duration(config.DefaultQueueConfig.BatchSendDeadline)/time.Second)
-		}
 		m.Start()
+
 		desiredShards := m.calculateDesiredShards()
+		shouldReshard := m.shouldReshard(desiredShards)
+
 		m.Stop()
-		if !c.reshard {
-			testutil.Assert(t, desiredShards == m.numShards, "expected calculateDesiredShards to not want to reshard, wants to change from %d to %d shards", m.numShards, desiredShards)
-		} else {
-			testutil.Assert(t, desiredShards != m.numShards, "expected calculateDesiredShards to want to reshard, wants to change from %d to %d shards", m.numShards, desiredShards)
-		}
+
+		testutil.Assert(t, shouldReshard == c.expectedToReshard, "shouldReshard set to %t, expected %t", shouldReshard, c.expectedToReshard)
```

Comment:
`testutil.Equals(t, c.expectedToReshard, shouldReshard)` rather than `Assert` will handle the print formatting for you in a consistent manner with other test functions.

Reply:
This is now a single assertion.

Reply:
Yep, thanks, the logic is much easier to follow than before!

Since you are now comparing two values usually we use `testutil.Equals(t, expected, actual)` instead of `testutil.Assert(t, expected == actual, "my message")` as the former provides some nice out of the box formatting.

Reply:
Done, thanks for the suggestion!

Sorry for the weird reply to the initial comment, you had made a different comment to simplify the two test assertions in the old file, which I changed and then did a rebase with force push.

It looks like Github didn't display the old comment any more (due to the force push) and you made this comment on exactly the same line.

I went on autopilot mode and just replied to this one as though it was the old one.

I'll be more careful in the future.